src/krb5/_creds_mit.pyx

```diff
@@ -48,3 +58,58 @@ def get_validated_creds(
     creds.needs_free = 1
 
     return creds
+
+def get_etype_info(
+    Context context not None,
+    Principal principal not None,
+    GetInitCredsOpt opt = None,
+) -> typing.Tuple["krb5_enctype", bytes, bytes]:
```

Comment:
For anything more complex I think we should return a named tuple instead of just a regular one. See https://github.com/jborean93/pykrb5/blob/81d823e9ec21b8511c4c348acc8d1fac3f2f1918/src/krb5/_creds.pyx#L499-L504 for an example of how it can be defined in the pyx and https://github.com/jborean93/pykrb5/blob/81d823e9ec21b8511c4c348acc8d1fac3f2f1918/src/krb5/_creds.pyi#L13-L17 how it is defined in the pyi.

A named tuple can still be used like a normal tuple when unpacking but it allows the caller to get the return values like normal class attributes.

Also based on the code the salt and s2k bytes can be Optional[bytes] (if `ENCTYPE_NULL` is set).

Reply:
I'll change that to a named tuple.

> Also based on the code the salt and s2k bytes can be Optional[bytes] (if `ENCTYPE_NULL` is set).

Yes, and I actually did that correctly in the `.pyi` file, but forgot to update the `.pyx` file.